### PR TITLE
scanner,preload: remove channel buffers

### PR DESCRIPTION
**Pull request by FiloSottile**
The fetches chan buffer can be removed without any impact as the
generator is fast, and has to block on the consumer anyway.

The big arbitrary jobs buffer feels misplaced hidden in a generic-use
Scanner object, it should be up to the application to decide if it wants to
queue or not. It also unavoidably completely fills when the consumer is
slower than the fetcher (like in preload), adding 6GB of memory usage.

The point of buffering in preload is to not ever have a submitter block
on a fetch. Assuming a fetch takes at most X times a submit, and
considering fetches are batched while submits are not, a buffer of
length X (for each submitter) will suffice to guarantee that. I picked
10 as a conservative X, but I suspect it might be < 1.

Notice how preload can make a more informed decision on queueing than
scanner, confirming the choice about the jobs buffer.

[Imported from #36]

### CODE REVIEW

File: scanner/scanner.go [hunk 1]
```diff
@@ -285,8 +285,8 @@ func (s *Scanner) Scan(ctx context.Context, foundCert func(*ct.LogEntry), foundP
 
 	ticker := time.NewTicker(time.Second)
 	startTime := time.Now()
-	fetches := make(chan fetchRange, 1000)
-	jobs := make(chan entryInfo, 100000)
+	fetches := make(chan fetchRange)
+	jobs := make(chan entryInfo)
```
Comment (daviddrysdale) [inline scanner/scanner.go]: How about keeping `jobs` still buffered, but make the size a `ScannerOptions` field?

Comment (FiloSottile) [reply scanner/scanner.go]: I am fairly convinced buffering belongs in the application, one layer up, not in the generator object.

Most of the work consuming entires from jobs is done by foundCert/foundPrecert so buffering can be implemented there if needed.

But of course, this is your codebase, I won't be offended if you push that change 😄 (GitHub lets you push to my branch) my main goal is to cut the 6GB of fixed buffer.

Comment (daviddrysdale) [reply scanner/scanner.go]: I'd rephrase as "control of buffering belongs in the application, not in the generator object" and then we'd be pretty much in agreement :-)

Comment (FiloSottile) [reply scanner/scanner.go]: Fair enough :) I can't push right now, so feel free to if you have time, otherwise I'll get to it this week.

Comment (FiloSottile) [reply scanner/scanner.go]: @daviddrysdale Done
